ddtrace/appsec/_iast/_taint_tracking/TaintTracking/TaintRange.cpp

```diff
@@ -92,13 +104,17 @@ PyObject*
 api_set_ranges_from_values(PyObject* self, PyObject* const* args, Py_ssize_t nargs)
 {
     bool result = false;
-    const char* result_error_msg =
-      "[IAST] Invalid number of params: pyobject_newid, len(pyobject), source_name, source_value, source_origin";
+    const char* result_error_msg = MSG_ERROR_N_PARAMS;
     PyObject* pyobject_n = nullptr;
 
     if (nargs == 5) {
         PyObject* tainted_object = args[0];
-        auto ctx_map = initializer->get_tainting_map();
+        TaintRangeMapType* tx_map = initializer->get_tainting_map();
+        if (not tx_map) {
+            py::set_error(PyExc_ValueError, MSG_ERROR_TAINT_MAP);
+            return nullptr;
```

Comment:
Shouldn't we return the original pyobject here, if I understand correctly, if IAST is not enabled for this request, we can end up here, and this would be setting to `None` any try to taint a pyobject